censys/asm/inventory.py

```diff
@@ -57,7 +62,24 @@ def search(
         if fields:
             args["fields"] = fields
 
-        return self._get(self.base_path, args=args)
+        page = 1
+        next_cursor = None
+        hits = []
+        resp = self._get(self.base_path, args=args)
+        next_cursor = resp.get("nextCursor")
+        hits.extend(resp.get("hits", []))
+        while next_cursor and (page == 0 or pages == -1 or page < pages):
```

Comment:
i had to reason through the fact that a non-200 response would not be caught here (since next_cursor would be `None`), but perhaps a comment could make it clearer that we effectively are short circuiting and returning the non-200 response as soon as we get it

Reply:
add a comment for context and removed `page == 0`, it was unnecessary